### ISSUE TITLE
`multiple_supertrait_upcastable` lint is maybe overzealous for auto traits

### ISSUE THREAD

**Issue by zachs18** [rust-lang/rust#151438]
I tried this code:

```rust
#![feature(multiple_supertrait_upcastable)]
#![warn(multiple_supertrait_upcastable)]

pub trait Foo: Send + Sync {}
```

I expected to see this happen: `Send` and `Sync` are auto traits, and thus do not (currently) take any extra space in the vtable for `dyn Foo`[^1], so the `multiple_supertrait_upcastable` lint should not fire. (similarly if `Foo` had one auto and one non-auto supertrait).

Instead, this happened: The `multiple_supertrait_upcastable` lint fires, since `Foo` has multiple supertraits (`Send` and `Sync`) ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=7783c2319f6d24a699221c8307107378)).

Tracking issue for `multiple_supertrait_upcastable`: #150833

This could be determined to not be a false positive, as the diagnostic is *technically* correct. However, the reason for the lint when it was introduced was:

> To support upcasting with multiple supertraits, we need to store multiple vtables and
this can result in extra space overhead, even if no code actually uses upcasting. This
lint allows users to identify when such scenarios occur and to decide whether the
additional overhead is justified

Under that reasoning, I think `Foo` should not be linted on.

There could also be an argument for not linting on empty non-auto supertraits, but that is less convincing IMO.

<details><summary>non-auto supertraits</summary>

There could also be an argument for not linting on `Bar` below, as `Super1` and `Super2` also do not take any extra vtable space. However, this is less convincing IMO, since they could semver-compatibly get new items with default implementations that would give them vtable entries, whereas auto traits like `Send`/`Sync` cannot (currently).

```Rust
pub trait Super1 {} // empty
pub trait Super2 {} // empty
pub trait Bar: Super1 + Super2 { /* ... */ }
```

</details>

### Meta

`rustc --version --verbose`:
```
rustc 1.95.0-nightly (d940e5684 2026-01-19)
binary: rustc
commit-hash: d940e56841ddcc05671ead99290e35ff2e98369f
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 21.1.8
```

(no backtrace)

[^1]: upcasting `dyn Foo` to `dyn Send` just re-uses the vtable since it has a common prefix of `drop_in_place, size, align`
